### PR TITLE
Fix some NumPy 2.x CI failures (cont.)

### PR DESCRIPTION
**Pull request by EarlMilktea**
Related to #8565 and #8690 .

This PR also partly fixes https://ci.preferred.jp/cupy.linux.cuda126/176161/ .

### Fixed

- `cupy_tests/statistics_tests/test_meanvar.py::TestMeanVar::test_mean_all_float64_dtype`
- `cupy_tests/testing_tests/test_helper.py::TestPackageRequirements::test_installed`
- `cupy_tests/testing_tests/test_helper.py::TestPackageRequirements::test_with_requires`
- `cupy_tests/random_tests/test_generator.py::TestRandint::test_randint_int64_1`
- `cupy_tests/math_tests/test_rounding.py::TestRounding::test_ceil`
- `cupy_tests/math_tests/test_rounding.py::TestRounding::test_floor`
- `cupy_tests/math_tests/test_rounding.py::TestRounding::test_trunc`
- `cupy_tests/math_tests/test_rounding.py::TestRounding::test_fix`

### CODE REVIEW

File: tests/cupy_tests/statistics_tests/test_meanvar.py [hunk 1]
```diff
@@ -217,7 +217,12 @@ def test_external_mean_axis(self, xp, dtype):
     @testing.for_all_dtypes(no_complex=True)
     @testing.numpy_cupy_allclose()
     def test_mean_all_float64_dtype(self, xp, dtype):
-        a = xp.full((2, 3, 4), 123456789, dtype=dtype)
+        fillvalue = 123456789
+        try:
+            dtype(fillvalue)
+        except OverflowError:
+            pytest.skip("fillvalue is out of range.")
+        a = xp.full((2, 3, 4), fillvalue, dtype=dtype)
```
Comment (asi1024) [inline tests/cupy_tests/statistics_tests/test_meanvar.py:225]:
```suggestion
        a = testing.shaped_arange((2, 3, 4), xp, dtype)
```

File: tests/cupy_tests/random_tests/test_generator.py [hunk 2]
```diff
@@ -969,11 +969,8 @@ def test_randint_float1(self):
     def test_randint_float2(self):
         self.generate(6.7, size=(2, 3))
 
-    @pytest.mark.xfail(numpy.__version__ < "2",
-                       reason='XXX: np 2.0: comparisons with OOB '
-                              'ints are broken in numpy < 2'
-                       )
     def test_randint_int64_1(self):
+        # FIXME: May have subtle bug in function signature determination
```
Comment (asi1024) [inline tests/cupy_tests/random_tests/test_generator.py:973]: There's still a bug in the comparison logic, but at least the bug in randint has been resolved. Could you please remove the comments here?

File: cupy/_math/rounding.py [hunk 3]
```diff
@@ -46,39 +46,110 @@ def round_(a, decimals=0, out=None):
...
-
-    .. seealso:: :data:`numpy.ceil`
-
-    ''', support_complex=False)
-
-
-trunc = ufunc.create_math_ufunc(
-    'trunc', 1, 'cupy_trunc',
-    '''Rounds each element of an array towards zero.
+floor = _core.create_ufunc(
```
Comment (asi1024) [inline cupy/_math/rounding.py:49]: How about defining `create_rounding_ufunc`?

```py
def create_rounding_ufunc(name, op, doc):
    return _core.create_ufunc(
        name,
        ("?->?", "b->b", "B->B", "h->h", "H->H", "i->i", "I->I",
         "l->l", "L->L", "q->q", "Q->Q",
         ("e->e", op), ("f->f", op), ("d->d", op)),
        "out0 = in0",
        doc=doc,
    )
```

File: tests/cupy_tests/random_tests/test_generator.py (hunk 2 above)
Comment (EarlMilktea) [reply tests/cupy_tests/random_tests/test_generator.py]: Removed in 2086313283cc272e58871c49a71dbaf4e928971e .

File: cupy/_math/rounding.py (hunk 3 above)
Comment (EarlMilktea) [reply cupy/_math/rounding.py]: Please check ff4e8a6d64d843a65272ec5775588e49627a0578 .

**Review by asi1024 — APPROVED** [review]
LGTM.

**Pull request merged**